# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/client/ClientCanvasRenderer.cpp
# Commit: dd00e2da3a0f
# Full Hash: dd00e2da3a0f11a7cc0af9e5ece21a339f4393eb
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 69cc1de6f754 -r dd00e2da3a0f gfx/layers/client/ClientCanvasRenderer.cpp
--- a/gfx/layers/client/ClientCanvasRenderer.cpp	Wed Jun 10 21:57:19 2020 +0000
+++ b/gfx/layers/client/ClientCanvasRenderer.cpp	Wed Jun 10 22:21:02 2020 +0000
@@ -17,35 +17,16 @@
 
 bool ClientCanvasRenderer::CreateCompositable() {
   if (!mCanvasClient) {
-    TextureFlags flags = TextureFlags::DEFAULT;
-    if (mOriginPos == gl::OriginPos::BottomLeft) {
-      flags |= TextureFlags::ORIGIN_BOTTOM_LEFT;
-    }
-
-    if (IsOpaque()) {
-      flags |= TextureFlags::IS_OPAQUE;
+    auto compositableFlags = TextureFlags::NO_FLAGS;
+    if (!mData.mIsAlphaPremult) {
+      // WR needs this flag marked on the compositable, not just the texture.
+      compositableFlags |= TextureFlags::NON_PREMULTIPLIED;
     }
-
-    if (!mIsAlphaPremultiplied) {
-      flags |= TextureFlags::NON_PREMULTIPLIED;
-    }
-
-    mCanvasClient = CanvasClient::CreateCanvasClient(GetCanvasClientType(),
-                                                     GetForwarder(), flags);
-    if (!mCanvasClient) {
-      return false;
-    }
+    mCanvasClient = new CanvasClient(GetForwarder(), compositableFlags);
 
     if (mLayer->HasShadow()) {
-      if (mAsyncRenderer) {
-        static_cast<CanvasClientBridge*>(mCanvasClient.get())->SetLayer(mLayer);
-      } else if (mOOPRenderer) {
-        static_cast<CanvasClientOOP*>(mCanvasClient.get())
-            ->SetLayer(mLayer, mOOPRenderer);
-      } else {
-        mCanvasClient->Connect();
-        GetForwarder()->AsLayerForwarder()->Attach(mCanvasClient, mLayer);
-      }
+      mCanvasClient->Connect();
+      GetForwarder()->AsLayerForwarder()->Attach(mCanvasClient, mLayer);
     }
   }
 